synapse/lib/view.py

```diff
@@ -0,0 +1,358 @@
+import asyncio
+import logging
+import itertools
+
+import synapse.exc as s_exc
+import synapse.common as s_common
+
+import synapse.lib.base as s_base
+import synapse.lib.coro as s_coro
+import synapse.lib.snap as s_snap
+import synapse.lib.trigger as s_trigger
+
+logger = logging.getLogger(__name__)
+
+class View(s_base.Base):
+    '''
+    A view represents a cortex as seen from a specific set of layers.
+
+    The view class is used to implement Copy-On-Write layers as well as
+    interact with a subset of the layers configured in a Cortex.
+    '''
+
+    async def __anit__(self, core, node):
+        '''
+        Async init the view.
+
+        Args:
+            core (Cortex):  The cortex that owns the view.
+            node (HiveNode): The hive node containing the view info.
+        '''
+        await s_base.Base.__anit__(self)
+
+        self.core = core
+
+        self.node = node
+        self.iden = node.name()
+
+        self.invalid = None
+        self.parent = None  # The view this view was forked from
+
+        self.info = await node.dict()
+        self.info.setdefault('owner', 'root')
+        self.info.setdefault('layers', ())
+
+        self.triggers = s_trigger.Triggers(self)
+
+        self.layers = []
+
+        for iden in self.info.get('layers'):
+
+            layr = core.layers.get(iden)
+
+            if layr is None:
+                self.invalid = iden
+                logger.warning('view %r has missing layer %r' % (self.iden, iden))
+                continue
+
+            if not self.layers and layr.readonly:
+                self.invalid = iden
+                raise s_exc.ReadOnlyLayer(mesg=f'First layer {iden} must not be read-only')
+
+            self.layers.append(layr)
+
+    async def eval(self, text, opts=None, user=None):
```

Comment:
I'd like to take this opportunity to clarify/unify eval / storm / stormstream APIs.  I think eval probably goes away.

Reply:
( as an internal function.  we can maintain the outer facing API )

Reply:
Let's talk about this.

Reply:
resolved via vox